### PR TITLE
[Chore] Setup Unit testing

### PR DESCRIPTION
**Pull request by tyrro**
## What happened 👀

Closes https://github.com/tyrro/react-survey/issues/6
Jest and React Testing Library are setup as unit testing utilities

## Insight 📝

N/A
 
## Proof Of Work 📹

image

### CODE REVIEW

File: pages/index.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+describe(Home, () => {
+  it('renders a heading', () => {
+    render(<Home />);
+
+    const heading = screen.getByRole('heading', {
+      name: /welcome to next\.js!/i,
+    });
+
+    expect(heading).toBeInTheDocument();
```
Comment (carryall) [inline pages/index.test.js:12]: If you agree on setting up the test id attribute on the [previous PR](https://github.com/tyrro/react-survey/pull/47#discussion_r831760337) maybe use it here too 💭 
We can also [configure the test id attribute](https://testing-library.com/docs/dom-testing-library/api-configuration/) so when we add the test id to the component we can use it on both UI and unit test 🙌🏼 

```suggestion
    const heading = screen.getByTestId(headingTestId);

    expect(heading).toBeVisible();
    expect(heading).toHaveTextContent(...);
```

**Review by carryall — COMMENTED** [review]
The same small question I added on the previous PR, any plan to add the code coverage report? If there's a plan maybe you'll want to merge the report from UI and unit test, you can take a look at the way we did on [the Varun project](https://github.com/nimblehq/varun-web/blob/develop/scripts/coverage-merge.js) 😄

**Review by malparty — APPROVED** [review]
LGTM with Pae's comments ;)

**Comment by tyrro** [comment]
> The same small question I added on the previous PR, any plan to add the code coverage report? If there's a plan maybe you'll want to merge the report from UI and unit test, you can take a look at the way we did on [the Varun project](https://github.com/nimblehq/varun-web/blob/develop/scripts/coverage-merge.js) 😄
>

done at https://github.com/tyrro/react-survey/pull/48/commits/453739d04c1070363c45776e745bf5c26a11668b 👍

File: package.json [hunk 2]
```diff
@@ -8,8 +8,17 @@
...
-    "cypress:run": "cypress run"
+    "cypress:run": "cypress run",
+    "test:coverage": "rm -f ./.nyc_output/out.json && yarn test --coverage --watchAll=false && yarn cypress:run && node ./scripts/coverage-merge.js && nyc report",
+    "test:coverage-merge": "node ./scripts/coverage-merge.js"
+  },
+  "jest": {
+    "collectCoverage": true,
+    "coverageReporters": [
+      "json"
+    ]
```
Comment (carryall) [inline package.json:21]: Can we move this to `jest.config.js`? I think it causes this warning when running the test 🤔 

https://user-images.githubusercontent.com/1772999/159621080-70ec5a58-b6a3-4e68-822a-204b52a6bb38.mov

File: pages/index.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { render, screen } from '@testing-library/react';
+import Home from '.';
+
+describe(Home, () => {
+  it('renders a heading', () => {
+    render(<Home />);
+
+    const heading = screen.getByTestId('welcome-text');
+    expect(heading).toBeVisible();
```
Comment (carryall) [inline pages/index.test.js:9]: How about adding some space here to separate the declaration and assertion?💡 

```suggestion
    const heading = screen.getByTestId('welcome-text');

    expect(heading).toBeVisible();
```

**Review by carryall — APPROVED** [review]
Only 2 small comments, the rest LGTM 🙌🏼

File: package.json (hunk 2 above)
Comment (tyrro) [reply package.json]: sure, done in https://github.com/tyrro/react-survey/pull/49

File: pages/index.test.js (hunk 3 above)
Comment (tyrro) [reply pages/index.test.js]: sure, done here https://github.com/tyrro/react-survey/pull/49 👍

**Pull request closed without merging**